Data/samples/DBLogger/src/DBLogger.cpp

```diff
@@ -0,0 +1,337 @@
+
+class DBLogger: public Application
+{
+public:
+	DBLogger():
+		_helpRequested(false),
+		_sqlNameGlob("*.sql")
+	{
+	}
+
+	~DBLogger()
+	{
+	}
+
+protected:
+	void initialize(Application& self)
+	{
+		loadConfiguration(); // load default configuration files, if present
+		Application::initialize(self);
+
+		Poco::Data::SQLite::Connector::registerConnector();
+
+		logger().information("conector: %s, cs: %s", _connector, _connectionString);
+
+		_dataSession = std::make_shared<Poco::Data::Session>(_connector, _connectionString);
+
+		(*_dataSession) << ("DROP TABLE IF EXISTS "s + _tableName), now;
+		const auto create {
+			"CREATE TABLE "s + _tableName +
+			" (Source VARCHAR, Name VARCHAR, ProcessId INTEGER, Thread VARCHAR,"s +
+			" ThreadId INTEGER, Priority INTEGER, Text VARCHAR, DateTime DATE)"s
+		};
+
+		(*_dataSession) << create, now;
+
+		// Initial scan of the directory to pick existing files
+		scanDirectory();
+
+		// Watcher to process files as they are put in the directory
+		_dirWatcher = std::make_shared<Poco::DirectoryWatcher>(_directory, Poco::DirectoryWatcher::DW_FILTER_ENABLE_ALL);
+
+		_dirWatcher->itemAdded += Poco::delegate(this, &DBLogger::onItemAdded);
+		_dirWatcher->itemRemoved += Poco::delegate(this, &DBLogger::onItemRemoved);
+		_dirWatcher->itemModified += Poco::delegate(this, &DBLogger::onItemModified);
+		_dirWatcher->itemMovedFrom += Poco::delegate(this, &DBLogger::onItemMovedFrom);
+		_dirWatcher->itemMovedTo += Poco::delegate(this, &DBLogger::onItemMovedTo);
+
+		// SQL channel to generate SQL files
+		_sqlChannel = new Poco::Data::SQLChannel();
+		_sqlChannel->setProperty(Poco::Data::SQLChannel::PROP_DIRECTORY, _directory);
+		_sqlChannel->setProperty(Poco::Data::SQLChannel::PROP_TABLE, _tableName);
+
+		_active = true;
+		_startTime.update();
+
+		_sqlSourceThread.startFunc(
+			[this]() { this->createMessages(); }
+		);
+
+		logger().information("Scanning started: %s", _dirWatcher->directory().absolutePath());
+	}
+
+	void uninitialize()
+	{
+		_sqlSourceThread.join();
+		_dirWatcher->suspendEvents();
+
+		logger().information(
+			"Created %z messages, processed %z messages in %Ld ms.",
+			_created, _processed, (_startTime.elapsed() / 1000)
+		);
+
+		Application::uninitialize();
+	}
+
+	void reinitialize(Application& self)
+	{
+		Application::reinitialize(self);
+	}
+
+	void defineOptions(OptionSet& options)
+	{
+		Application::defineOptions(options);
+
+		options.addOption(
+			Option("help", "h", "display help information on command line arguments")
+				.required(false)
+				.repeatable(false)
+				.callback(OptionCallback<DBLogger>(this, &DBLogger::handleHelp)));
+
+		options.addOption(
+			Option("?", "?", "display help information on command line arguments")
+				.required(false)
+				.repeatable(false)
+				.callback(OptionCallback<DBLogger>(this, &DBLogger::handleHelp)));
+
+		options.addOption(
+			Option("dir", "d", "directory path to scan for SQL log files")
```

Comment:
these options should also be configurable from `DBLogger.properties` file

Reply:
Implemented changes to provide options via configuration file and command line.